File: 1497-check-if-array-pairs-are-divisible-by-k/1497-check-if-array-pairs-are-divisible-by-k.java
class Solution {
    public boolean canArrange(int[] arr, int k) {
        HashMap<Integer,Integer> map=new HashMap<>();
        for(int i=0;i<arr.length;i++){
            int rem=((arr[i]%k)+k)%k;
            map.put(rem,map.getOrDefault(rem,0)+1);
        }
        for(int i=0;i<arr.length;i++){
            int r=arr[i]%k;
            if((r==0)){
                if(map.get(r)%2!=0){
                    return false;
                }
            }else if(2*r==k){
                if(map.get(r)%2!=0){
                    return false;
                }
            }
            else{
                int fq=map.getOrDefault(r,0);
                int ofq=map.getOrDefault(k-r,0);
                if(fq!=ofq){
                    return false;
                }
            }
        }
        return true;
    }
}

File: 1497-check-if-array-pairs-are-divisible-by-k/1497-check-if-array-pairs-are-divisible-by-k.ts
var canArrange = function(arr, k) {
    const freq = new Array(k).fill(0);
        
        for (const num of arr) {
            let rem = num % k;
            if (rem < 0) {
                rem += k;
            }
            freq[rem]++;
        }
        
        if (freq[0] % 2 !== 0) {
            return false;
        }
        
        for (let i = 1; i <= Math.floor(k / 2); i++) {
            if (freq[i] !== freq[k - i]) {
                return false;
            }
        }
        
        return true;
};